### ISSUE TITLE
Build failures when link MFMG/no CUDA against DealII/CUDA

### ISSUE THREAD

**Issue by aprokop** [ORNL-CEES/mfmg#65]
```
<snip>
/usr/local/cuda-8.0/lib64/libcusolver.so: undefined reference to `GOMP_parallel_start'
/usr/local/cuda-8.0/lib64/libcusolver.so: undefined reference to `GOMP_critical_end'
<snip>
```
`-fopenmp` flag is present in the link, and is after `cusolver`. DealII was installed through spack. This does not happen when DealII is installed independently.

Enabling CUDA in MFMG leads to successful linking.

**Comment by Rombur** [comment]
Yeah there is a bug with the current development version of dealii where `-fopenmp` is not passed correctly when cusolver is used https://cdash.kyomu.43-1.org/testDetails.php?test=8527129&build=1390 I am hoping to fix it this week end.

**Comment by masterleinad** [comment]
I am seeing similar issues if `deal.II` is built with `CUDA` support, but `mfmg` isn't.
